src/indexed_crate.rs

```diff
@@ -148,3 +156,73 @@ fn collect_public_items<'a>(
         Visibility::Crate | Visibility::Restricted { .. } => {}
     }
 }
+
+fn create_dummy_trait_items(crate_: &Crate) -> HashMap<Id, Item> {
+    let paths = crate_
+        .index
+        .values()
+        .map(|item| &item.inner)
+        .filter_map(|item_enum| match item_enum {
+            rustdoc_types::ItemEnum::Impl(impl_) => Some(impl_),
+            _ => None,
+        })
+        .filter_map(|impl_| impl_.trait_.as_ref());
+
+    // Limiting the creation of dummy traits to only those that are used by the lints.
+    // There are other foreign traits and it is not obvious how the dummy traits
+    // should look like for them.
```

Comment:
But then, what happens when a user derives a trait not from the `std` crates, but from another foreign crate? The `derive_trait_impl_removed` lint would not work.

Reply:
Mentioned in https://github.com/obi1kenobi/cargo-semver-checks/issues/256.